README.md

```diff
@@ -214,6 +214,46 @@ The hander life cycle can be illustrated as:
 or optionally,
 
   `.start` -> `#before_consume` -> `#around_consume` [ `#consume` ] -> `.stop`
+  
+#### Batch Consumption
+
+In addition to the regular handler, Phobos provides a `BatchHandler`. The
+basic ideas are identical, except that instead of being passed a single message
+at a time, the `BatchHandler` is passed a batch of messages. All methods
+follow the same pattern as the regular handler except that they each
+end in `_batch` and are passed an array of `Phobos::BatchMessage`s instead
```

Comment:
I am a bit torn in the decision to suffix with `_batch` instead of following the established Phobos standard which would be `before_consume`, `around_consume`, and `consume`. One the one hand I like that it is explicit and communicative to the user. But maybe it is enough that they have included the BatchHandler and the context they are in is already "batch" mode?
The advantage of not introducing the suffix is less complexity on the maintainer side, easier documentation, the dots feel a little bit more connected...? WDYT @jasondoc3 , maybe you want to join the discussion?

Reply:
Let us consider the Phobos framework from the `phobos.yml` configuration file perspective... Here, there is a mapping from topic to consumer with a configuration stating the preferred method of delivery which can be `batch`, `inline_batch` and `message`. So, for a new user of Phobos, I would recommend that if `batch` or `message` is chosen, the consumer should include only `Phobos::Handler`, and if `inline_batch` is chosen the consumer should include only `Phobos::BatchHandler`. This is not documented in the README but it is in the phobos.yml example configuration:

```
    #   - `inline_batch` will use Phobos::BatchHandler instead of Phobos::Handler to pass the entire batch to your application
    #     at once, as opposed to `batch` which will still use the standard `Phobos::Handler` to process each message
    #     individually.
```

Having this perspective it seems like an irregular use case to have a consumer that in spite of this is including both handlers, and introducing some extra configuration (not covered by Phobos framework) trying to figure out if it should use `consume` or `consume_batch`. It sounds like it is making it more complicated than necessary.

Have you investigated the possibility to put your application's consumer logic somewhere else? I would really like to hear more details on what the pain point is that the Phobos framework cannot solve for you out of the box. I have limited insight in other peoples' Phobos implementations, and any glance of it is valued.

To be completely transparent with you I am asking this because I want to have as low friction as possible in the "core" of Phobos, and if the show stopper is this special case in your app, then maybe we can solve that together 🙏🏼.

Reply:
I found a counter argument to myself 🤔

If we went ahead and did what I argue for here, then the only difference between `Handler` and `BatchHandler` would be pluralisation of the parameters, e.g. `payload `vs `payloads`. Which could mean we might not even need or want `BatchHandler` at all. You simply include `Phobos::Handler` in either case and just make sure you know which mode of operation you are in. Which, in fact, could potentially be a bit vague/unclear as configuration change would require implementation changes. Going from `batch` to `batch_inline` would break the consumer in a not so nice way.

So if we KEEP the proposed suffix, then going from `batch` to `batch_inline` would still break the consumer, but in a much nicer way. It should suddenly fail with `undefined method :before_consume_batch` which would tell the user to start thinking in batches instead of individual messages.

A small thing like this could make this handler a bit more special, and should make configuration mistakes less common.

Reply:
Yep, exactly. My use case is that I have a single "base consumer" which can be subclassed for different schemas or different purposes. The base consumer would implement both the batch and the single methods, and depending on phobos.yml, the subclasses would just work(tm) without having to know the difference.

Essentially this is the difference between Java-style method overloading:
```java
  public void doSomething(int val);
  public void doSomething(String val);
```

and the JavaScript way:
```javascript
  function doSomething(val) {
    if (typeof(val) == 'int') { ...}
    else { ... }
}
```
I find the Java way much more explicit and an easier way to split functionality.

Reply:
I'll update the README though!